### ISSUE TITLE
v8 v7.7 rolling-origin drift-adaptive action-value policy

### ISSUE THREAD

**Issue by phead198708** [phead198708/BiGan#240]
## Goal

Create one drift-adaptive v8 candidate that must be no worse than frozen v6.7 on a same-market historical prequential replay before any new data collection begins.

This is a development screen only. It is not promotion evidence.

## Why this follows #239

#239 passed inclusive non-inferiority only by selecting `KEEP_V6_7` everywhere:

- exact-90 candidate PnL: `+0.7030`
- exact-90 v6.7 PnL: `+0.7030`
- total/LWR deltas: `0 / 0`
- action differences: `0`

Fixed 30/60/all recency experts did not produce an eligible actionable policy. This issue tests one generic drift-adaptive update mechanism rather than another result-selected window grid.

The #239 outer-OOF action/PnL rows and #238 side/action attribution diagnostics are forbidden design/tuning inputs.

## Candidate

`rolling_origin_drift_adaptive_action_value_v7_7`

Allowed decisions:

- `KEEP_V6_7`
- `SWITCH_SAME_DECISION_SBC`
- `VETO_TO_NO_TRADE`

Never activate a v6.7 `NO_TRADE` market. No HTS, side quota, alternative timestamp, or more than one bet per market.

## Historical stream

Seed model:

- existing canonical 134-market runtime-aligned historical corpus

Rolling-origin development stream:

- the consumed, closed #238 exact-120 markets
- chronological market order
- for market t, features and targets from market t are unavailable to its prediction
- after prediction is frozen, its official read-only settled target may enter the training history for t+1
- all 120 predictions are therefore strictly prior-only and prequential

The #238 stream is now historical development data, not an unseen holdout. Its replay cannot support promotion.

## Single preregistered model

Use the existing canonical decision-time features and frozen v7.4 XGBoost parameters.

Fit one joint SBC action-value model:

- training rows: both same-decision SBC sides
- target: runtime-aligned after-cost net return per contract
- sample weights: deterministic exponential decay with fixed half-life 60 markets
- no side-specific quota or hard minimum
- no hyperparameter/window/profile search

Decision rule with fixed `0.025` buffer:

1. Compute predicted baseline and same-decision opposite SBC returns.
2. Switch only if opposite predicted return is positive and exceeds baseline by at least `0.025`.
3. Veto to `NO_TRADE` only if predicted baseline return is below `-0.025` and opposite is not eligible.
4. Otherwise retain v6.7.

The buffer and half-life are frozen before implementation and are not selected from #238 outcomes.

## Historical prequential hard gate

On the exact same 120 #238 markets:

- candidate minus v6.7 total fixed-size after-cost PnL: `>= 0`
- candidate minus v6.7 largest-winner-removed PnL: `>= 0`
- equality passes
- exact v6.7 identity reconciliation
- exact 120 market support
- at least 3 action differences for collection actionability
- final fitted policy must be decision-time safe

If PnL is equal but action differences are zero, non-inferiority passes but collection remains blocked.

## Leakage controls

- no #239 outer OOF rows, metrics, or selected-profile results accepted as inputs
- no #238 side/action/PnL attribution used for features, thresholds, weights, or rules
- each #238 market target is opened only after its prediction is frozen
- no future market target used for earlier prediction
- target timestamps must be after market close
- feature max-input timestamp must be <= decision timestamp
- replay outcome is not used to rerun or tune this candidate

## Collection

Only after historical prequential non-inferiority and actionability pass:

- freeze the final model and exact lineage hashes
- collect a bounded 12-market strictly later outcome-blind canary
- verify target-free action differences and full-guard compatibility
- do not open canary outcomes in this issue

If the historical gate fails or produces fewer than 3 differences, stop before collection.

## Safety

- no paper/live unlock
- no Polymarket writes
- no wallet signing
- no capital at risk
- no source/O score mutation
- no promotion/freeze/handoff/#134/#146 unlock
- any future promotion still requires a separately preregistered strictly later unseen PnL holdout

**Comment by phead198708** [comment]
## Historical prequential gate passed; target-free canary authorized

Frozen implementation: `0bbf1b9d62dbea08ef6d4683acc12474d8fd1678`

Profile SHA-256: `133a78cf5af33cef1bfb4e6d6f5de1771e153c2450d9226282648f4232bf006a`

Authoritative run: `v7-7-rolling-origin-historical-prequential-lineage-corrected-20260723T000000Z`

### Exact same-120 result

- markets: `120`
- candidate selected bets: `110`
- v6.7 selected bets: `120`
- candidate fixed-size after-cost PnL: `+0.4415000000000001`
- v6.7 fixed-size after-cost PnL: `-0.09499999999999985`
- total PnL delta: `+0.5365`
- largest-winner-removed PnL delta: `+0.5365`
- policy differences: `35`
- decision distribution: `KEEP_V6_7=85, SWITCH_SAME_DECISION_SBC=25, VETO_TO_NO_TRADE=10`
- inclusive historical non-inferiority gate passed: `true`
- historical actionability passed: `true`
- target-free canary collection allowed: `true`
- target-free canary started: `false`

Every stream prediction was frozen before that market's target loader ran. Current-market targets were unavailable to current prediction; only prior closed-market targets entered the next fit. #239 OOF artifacts and #238 side/action attribution were not accepted as tuning inputs. This consumed stream is historical development evidence only, not promotion evidence.

### Lineage correction disclosure

An initial output used a mistyped metadata-only implementation SHA (`0bbf1b9e...` instead of `0bbf1b9d...`) and is invalid. It was deterministically replayed with the correct frozen commit and no code/profile/model/threshold/data change. Prediction rows, runtime targets, candidate rows, baseline rows, leakage report, and PnL report are byte-identical between attempts. Only model/manifest metadata hashes differ.

### Authoritative artifact hashes

- manifest: `49a409fb6720e12fdf7e3215223f39e05ebf901e4e85c42394d25b9f5d114c27`
- model: `29cad5c65feb9608b9ecf333fac190fba48617ca20af80f303b01568bfbfad40`
- report JSON: `75d6d2d7f4b387d3814cb0ed1e5cd998ec49006a23cfb0a1344f1ccc0ea554e9`
- report Markdown: `22e1c762384d4cd678dbbb9b32bfeb93666ec65bc8eef880f91c6fee2328952d`
- leakage audit: `41e07bd49ce5fa2572e97fad30698d8271f704b146ac4ec8ceacebff5b39f9f3`
- prequential policy rows: `8549b64eb47e5cfa9227f3164793f1f4bab33efaddc31ec3d7922d6494adf20c`
- loaded runtime targets: `34e04e066c8d58bd50f4844f135794810bd4e34a8d108be1194591889cbe61ba`
- candidate rows: `b60012fc679de4a5ecdbdc5289504b10f023d51a5f9a840779c6e56599c2ef1d`
- v6.7 baseline rows: `82acc593d57c37c77da866f989487273865c3501ab39aa8fe346aa15cedb74df`

### Safety

`paper_candidate_allowed=false`, `live_trading_enabled=false`, `capital_at_risk=false`, writes/wallet/handoff/source/freeze/promotion/#134/#146 all remain false. The next step is a bounded 12-market strictly-later outcome-blind canary; no canary outcomes may be opened in #240.

**Comment by phead198708** [comment]
## Target-free canary contract frozen before collection

Implementation/plan commit: `da4ae5f409e32583de83898d543600308c35de9e`

Canary plan SHA-256: `74e9a7e38f837982b91bfa19e473cb94cd575917097796f2e4d1686b13e7e825`

Frozen collection contract:

- historical comparison remains inclusive non-inferiority (`candidate - v6.7 >= 0`; equality passes)
- historical gate/actionability already passed (`+0.5365` total and LWR delta; 35 action differences)
- select the earliest exact `12` quality-valid strictly-later/disjoint markets
- bounded scan cap: `18` attempted markets
- minimum market start is strictly after `1784755200000` (`2026-07-22T21:20:00Z`)
- minimum target-free actionability: at least `1` guard-accepted v7.7 policy difference
- no UP/DOWN side quota
- full v6.7 execution guard unchanged
- no outcomes, resolution, labels, or PnL may be opened
- no model/score/buffer/threshold tuning after collection

The raw collector remains candidate-agnostic. Frozen v6.2/v6.7 source scoring is applied only after each bounded batch is sealed. The v7.7 consumer preserves v6.7 `NO_TRADE`, uses only same-decision SBC pairs, and fails closed on any forbidden target field.

Validation before collection: 36 related tests passed; full v8 Ruff passed; staged and unstaged diff checks passed.

Safety remains unchanged: paper/live/write/wallet/capital/handoff/source/freeze/promotion/#134/#146 are all false. Passing this canary would authorize only a separately preregistered future unseen PnL holdout, not paper/live.

**Comment by phead198708** [comment]
## v7.7 target-free future canary: PASS

The preregistered target-free canary has completed on the earliest exact 12 quality-valid strictly-later/disjoint markets. No outcome, settlement, label, or PnL field was opened.

### Historical gate semantics

The historical comparison is inclusive non-inferiority, as requested:

```text
candidate fixed-size after-cost PnL >= v6.7 fixed-size after-cost PnL
```

Equality passes; v7.7 is not required to strictly beat v6.7. On the frozen historical exact-120 corpus, v7.7 was above v6.7 (`+0.4415` vs `-0.0950`, delta `+0.5365`).

### Collection audit

- collector commit: `42e4d9a8aa6a2e06854fd35e4d47dd71dc5062b1`
- implementation commit: `da4ae5f409e32583de83898d543600308c35de9e`
- canary plan SHA-256: `74e9a7e38f837982b91bfa19e473cb94cd575917097796f2e4d1686b13e7e825`
- bounded batch: `persistent-outcome-blind-000001-20260722T212431Z`
- attempted/indexed/quality-valid: `18 / 18 / 18`
- unique markets/slugs: `18 / 18`
- first market start: `1784755500000`, strictly greater than frozen boundary `1784755200000`
- raw artifact descriptors verified: `108`
- raw hash mismatches: `0`
- raw row-count mismatches: `0`
- raw resolution rows: `0`
- feature timestamp causality violations: `0`
- prior market overlap: `0`
- labels/outcomes/PnL opened: `false`
- resolution provider called: `false`
- settlement finalizer started: `false`
- training corpus export attempted: `false`
- index hash-chain validation: `true`

Hashes:

- batch summary: `6fc37e15b7a4a9e7a16ec46f8b62ea49fa5e3b26406d3b82c73422b2f8f81b34`
- append-only index: `8ba24aa9d0f36adae574f52394018a8a230f04cd179f36d890b6b8210e82654c`
- last index entry: `681d59d612dda6f2cd551b48ea26d03160fb8e4c4246dc99c304fbf2080f2bf2`
- batch manifest: `284bf75fbb74481539ca88106ec816f572393d72b20666b451ac811afc7663c9`
- development canary manifest: `a216aca4afe4c8ac90f969d7c575abdaad05a4e01fd1d2252f888842abb96fed`
- v6.2 source canary manifest: `2a2ef288f0cd1d17513816a192c1d322fcad1f78517e47d241cac1c6bc7b102a`

Operational disclosure: the `launchctl submit` job attempted to restart after the bounded command completed. It was stopped and removed immediately. It left only two empty, unindexed batch-2 directories (`0B`, no files). The authoritative service state and index remain batch 1 / 18 rows; batch 2 is diagnostic-only and is excluded from freezing, scoring, and support.

### Target-free actionability gate

Run: `v7-7-target-free-canary-20260722T230403Z`

- selected market count: `12`
- v6.7 positive guard-compatible market count: `11`
- v7.7 policy decisions: `KEEP_V6_7=9`, `SWITCH_SAME_DECISION_SBC=2`, `VETO_TO_NO_TRADE=1`
- selected actions: `BUY_UP_SBC=1`, `BUY_DOWN_SBC=9`, `NO_TRADE=2`
- full-guard accepted bets: `10`
- full-guard accepted policy differences vs v6.7: `2`
- strictly-later violations: `0`
- prior-market overlaps: `0`
- feature-causality violations: `0`
- target-free canary passed: `true`
- blocking reason codes: `[]`

The two accepted action differences are same-decision SBC side switches for markets:

- `0xc0e1569c0a5e154f2676f1825ea79387be62240e29ad12a91d32cc718f441c9a`
- `0xf0214f9b17fb685b7961e4d7d3ebe8307e62104259a92645543b219d8b70597d`

Artifact hashes:

- report: `6a68ab23a21c0142869e24f44667fb4c43aa0a3b64a397236e679c82f82c5e46`
- manifest: `3289f47301f42322656d33e981c459d08a6fd04e3b708e5ce49e0f123d61504f`
- decisions: `63abdeea2cf18e0a771bea164326dea5ffdf1bd9a8d6083305e33c20665aae8c`
- guard replay: `8af17a3907735e2a030eab1c44ead1b135cc8dbf612ea431953b6633a016c105`
- guard accepted rows: `0b5de82bc534510941c85263b13303760f34363b763b3f97cfccc2a775774bd5`
- canonical SBC rows: `78cc6871d6df06ff5a1c4a987e757c43e7191581b72c2494fbf77face7ecfdfb`

Validation: `11 passed`; Ruff passed; both diff checks passed.

### Decision

The target-free canary authorizes planning a separate strictly-later future-unseen PnL holdout. It is not promotion evidence and it does not unlock paper/live/handoff:

- `future_unseen_pnl_holdout_authorized=true`
- `future_unseen_pnl_holdout_started=false`
- `source_model_candidate_eligible=false`
- `freeze_ready=false`
- `promotion_evidence_eligible=false`
- `paper_candidate_allowed=false`
- `v8_execution_handoff_allowed=false`
- `capital_at_risk=false`
- `#134_resume_allowed=false`
- `#146_start_allowed=false`

**Comment by phead198708** [comment]
## Downstream future holdout completed

The frozen v7.7 candidate advanced through the target-free canary and was evaluated exactly once in #241 on 120 strictly-later unseen markets.

Result:

- v7.7 after-cost PnL: `-0.585700`
- v6.7 after-cost PnL: `-0.698600`
- comparative delta: `+0.112900`
- largest-winner-removed comparative delta: `+0.112900`
- comparative non-inferiority: passed
- absolute positive-PnL gate: failed
- final blocker: `candidate_total_after_cost_pnl_not_positive`

Therefore v7.7 is terminally rejected for paper/promotion despite improving over v6.7. Full single-use evidence: https://github.com/phead198708/BiGan/issues/241#issuecomment-5058216687

No future result will be used to refit or tune v7.7; all paper/live/handoff/promotion flags remain blocked.
